firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Protocols/CRCElements.tpp

```diff
@@ -50,7 +50,8 @@ IndexStatus CRCElement<PayloadBuffer>::parse(const Util::ByteVector<input_size>
       !std::is_const<PayloadBuffer>::value,
       "Parse method unavailable for CRCElements with const PayloadBuffer type");
 
-  if (input_buffer.size() < CRCElementHeaderProps::header_size) {
+  if (input_buffer.size() < CRCElementHeaderProps::header_size ||
+      input_buffer.size() > (payload_.max_size() + CRCElementHeaderProps::header_size)) {
```

Comment:
Let's see if we can do a static_assert on input_buffer.max_size() vs. payload_.max_size() + CRCElementHeaderProps::header_size for checking whether the input_buffer is small enough. This way we don't need to define a test checking the sizes of the buffers, we just need to describe in the documentation for the method.

Reply:
For the current condition , we need to use `payload_.max_size()` , So if we need to access `payload_` that means we are accessing `this` which is not a compile time constant and static_assert will fail,
we could make max_size() method static to solve aforementioned issue. what do you suggest we do @ethanjli ?

places where max_size() method is used: 
- https://github.com/pez-globo/pufferfish-software/blob/8c73b599d3da4b3df112f22a7685385bec57327c/firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Serial/FDO2/Commands.cpp#L327 (This will fail if we make it static)
- https://github.com/pez-globo/pufferfish-software/blob/8c73b599d3da4b3df112f22a7685385bec57327c/firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Backend/Frames.tpp#L26

Reply:
Ah, good catch. If we make `Vector::max_size` be static, can we use `PayloadBuffer::max_size()` instead of `payload_.max_size()`? If so, let's do that.

Reply:
Changes to make:
FrameProps: using encoded_max_size = payload_max_size + 1
FrameProps: using EncodedBuffer = Util::ByteVector<encoded_max_size>
FrameProps: using PayloadBuffer = Util::ByteVector<payload_max_size>
Frames.h: using FrameChunkSplitter = Protocols::ChunkSplitter<FrameProps::encoded_max_size>;
FrameReceiver::output: FrameProps::EncodedBuffer temp_buffer;
FrameReceiver::output: FrameProps::PayloadBuffer &output_buffer
FrameSender::transform: const FrameProps::PayloadBuffer &input_buffer, FrameProps::ChunkBuffer &output_buffer
BackendReceiver::output: FrameProps::PayloadBuffer temp_buffer1;

Reply:
With these changes, I'll test on the Nucleo board before final approval.

Reply:
> using FrameChunkSplitter = Protocols::ChunkSplitterFrameProps::encoded_max_size;

I was wondering by not using chunk_max_size are we enforcing the `delimiter `to be in the `254 bytes` itself and not `254 non-delimiter bytes+ delimiter byte + ecnoded byte`.

Reply:
commit for the changes discussed: https://github.com/pez-globo/pufferfish-software/pull/281/commits/d8ffd6b5d066fe8fbe8b278cfa3c8958eba843cc

Reply:
It seems to run fine on the Nucleo board!